dvc/repo/get.py

```diff
@@ -66,31 +64,40 @@ def get(url, path, out=None, rev=None):
             # the same cache file might be used a few times in a directory.
             repo.cache.local.cache_types = ["reflink", "hardlink", "copy"]
 
-            output = None
-            output_error = None
-
             try:
                 output = repo.find_out_by_relpath(path)
-            except OutputNotFoundError as ex:
-                output_error = ex
-
-            is_git_file = output_error and not os.path.isabs(path)
-            is_not_cached = output and not output.use_cache
+            except OutputNotFoundError:
+                output = None
 
-            if is_git_file or is_not_cached:
-                _copy_git_file(repo, path, out, url)
-                return
+            if output and output.use_cache:
+                _get_cached(repo, output, out)
+            else:
+                # Either an uncached out with absolute path or a user error
+                if os.path.isabs(path):
+                    raise FileNotFoundError
```

Comment:
Also need a message here.

Reply:
That would be wrapped info PathMissingError, see notes on whether we should distinguish these cases.